syft/internal/fileresolver/temp.go

```diff
@@ -0,0 +1,30 @@
+package fileresolver
+
+import (
+	"fmt"
+
+	stereoscopeFile "github.com/anchore/stereoscope/pkg/file"
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/file"
+)
+
+func UseTempResolver[T any](fn func(r file.WritableResolver) (T, error)) (T, error) {
+	var t T
+	tmpDirGen := stereoscopeFile.NewTempDirGenerator("temp-resolver")
```

Comment:
nit: the name should be attributable to `syft` as a prefix

Reply:
Removed this function.